### PR TITLE
Change config to allow multiple hostDir and mountDir configs

### PR DESCRIPTION
**Pull request by ddysher**
First step for https://github.com/kubernetes-incubator/external-storage/issues/131

@msau42

### CODE REVIEW

File: local-volume/provisioner/pkg/deleter/deleter_test.go [hunk 1]
```diff
@@ -171,8 +172,9 @@ func testSetup(t *testing.T, config *testConfig) *Deleter {
 
 	config.apiUtil = util.NewFakeAPIUtil(config.apiShouldFail, config.cache)
 	userConfig := &common.UserConfig{
-		MountDir: testMountDir,
-		HostDir:  testHostDir,
+		DiscoveryMap: map[string]common.MountConfig{
+			"sc1": common.MountConfig{HostDir: testHostDir + "/test-dir", MountDir: testMountDir + "/test-dir"},
```
Comment (msau42) [inline local-volume/provisioner/pkg/deleter/deleter_test.go]: nit: can you put HostDir and MountDir on separate lines to make it more readable?

File: local-volume/provisioner/pkg/discovery/discovery_test.go [hunk 2]
```diff
@@ -46,9 +46,9 @@ var testNode = &v1.Node{
 	},
 }
 
-var scMapping = map[string]string{
-	"sc1": "dir1",
-	"sc2": "dir2",
+var scMapping = map[string]common.MountConfig{
+	"sc1": common.MountConfig{HostDir: testHostDir + "/dir1", MountDir: testMountDir + "/dir1"},
```
Comment (msau42) [inline local-volume/provisioner/pkg/discovery/discovery_test.go]: nit: same thing, separate lines

**Comment by ddysher** [comment]
@msau42 comment addressed and rebased to head, PTAL

**Pull request merged**